spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageHandler.java

```diff
@@ -71,13 +86,19 @@ public final void handleMessage(Message<?> message) {
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug(this + " received message: " + message);
 		}
+		long start = 0;
 		try {
 			if (message != null && this.shouldTrack) {
 				message = MessageHistory.write(message, this, this.getMessageBuilderFactory());
 			}
+			if (this.countsEnabled) {
+				this.handlerMetrics.beforeHandle(message);
```

Comment:
`start` should be adjusted to the `System.currentTimeMillis()` here.
Otherwise in the `SimpleMessageHandlerMetrics` `this.duration.append(System.currentTimeMillis() - 0);` is very big value :smile: